File: HelloWorld-Java/src/HelloWorld/HelloWorld.java
package HelloWorld;

/**
 * This class covers "Hello World"
 * <p>
 * Only has the main method
 * 
 */

public class HelloWorld {

    /**
     * Starts the main application
     * <p>
     * Call another method -SayHelloWorld-
     * <p>
     * 
     * @param args a string array containing all the arguments from the command line
     */

    public static void main(String[] args) {
	SayHelloWorld(args);
	// System.console().readLine();
    }

    /**
     * This method prints "Hello World" to the console
     * <p>
     * This method loops a number (Max) times
     *
     * @param args a string array containing all the arguments from the command line
     * 
     * @return value indicating number of times the method looped
     */

    public static int SayHelloWorld(String[] args) throws NumberFormatException, NullPointerException, ArrayIndexOutOfBoundsException {

	Integer i = 1;
	Integer Max = 3;

	int firstArg;
	try {
	    // firstArg = Integer.parseInt(args[0]);
	    firstArg = Integer.parseUnsignedInt(args[0]);

	    System.err.println("VALID: argument <" + args[0] + "> used");
	    System.err.println("  ");

	} catch (NumberFormatException e) {
	    System.err.println("Argument <" + args[0] + "> must be a Positive Integer");
	    System.err.println("  Using DEFAULT value: " + Max + " ");
	    System.err.println("  ");

	    firstArg = Max;

	    // throw new NumberFormatException("Argument <" + args[0] + "> must be a
	    // Positive Integer");

	} catch (NullPointerException e) {
	    System.err.println("MISSING ARGUMENT: Argument must be a Positive Integer");
	    System.err.println("  Using DEFAULT value: " + Max + " ");
	    System.err.println("  ");

	    firstArg = Max;

	    // throw new NullPointerException("MISSING ARGUMENT: Argument must be a Positive
	    // Integer");

	} catch (ArrayIndexOutOfBoundsException e) {
	    System.err.println("NO ARGUMENT: Argument must be a Positive Integer");
	    System.err.println("  Using DEFAULT value: " + Max + " ");
	    System.err.println("  ");

	    firstArg = Max;

	    // throw new ArrayIndexOutOfBoundsException("NO ARGUMENT: Argument must be a Positive
	    // Integer");

	} catch (Exception e) {

		if (args.length == 0) {
			System.err.println("ERROR: argument <" + "missing"  + "> caused an " + e + " error!");
	    } else {
			System.err.println("ERROR: argument <" + args[0] + "> caused an " + e + " error!");
	    }
	    	
	    System.err.println("  Using DEFAULT value: " + Max + " ");
	    System.err.println("  ");

	    firstArg = Max;
	}

	Max = firstArg;

	while (i <= Max) {
	    System.out.println("Hello World ! " + i++ + " of " + Max);
	}
	System.out.println(" ");

	return --i;
    }
}

File: HelloWorld-Csharp/HelloWorld/HelloWorld/HelloWorld.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HelloWorld
{
    public class HelloWorld
    {
        static void Main(String[] args)
        {
            SayHelloWorld(args);
            Console.ReadLine();
        }

        public static int SayHelloWorld(String[] args)
        {

            int i = 1;
            int Max = 3;

            int firstArg = 0;

            try
            {
                firstArg = (int)uint.Parse(args[0]);

                Console.WriteLine("VALID: argument <" + args[0] + "> used");
                Console.WriteLine("  ");
            }
            catch (OverflowException)
            {
                Console.WriteLine("Argument <" + args[0] + "> must be a Positive Integer");
                Console.WriteLine("  Using DEFAULT value: " + Max + " ");
                Console.WriteLine("  ");

                firstArg = Max;
            }
            catch (FormatException)
            {
                Console.WriteLine("Argument <" + args[0] + "> must be a Positive Integer");
                Console.WriteLine("  Setting DEFAULT value: " + Max + " ");
                Console.WriteLine("  ");

                firstArg = Max;
            }
            catch (IndexOutOfRangeException)
            {
                Console.WriteLine("MISSING ARGUMENT: Argument must be a Positive Integer");
                Console.WriteLine("  Using DEFAULT value: " + Max + " ");
                Console.WriteLine("  ");

                firstArg = Max;
            }
            catch (NullReferenceException)
            {
                Console.WriteLine("MISSING ARGUMENT: Argument must be a Positive Integer");
                Console.WriteLine("  Using DEFAULT value: " + Max + " ");
                Console.WriteLine("  ");

                firstArg = Max;
            }
            catch (Exception e)
            {
                Console.WriteLine("ERROR: argument <" + args[0] + "> caused an " + e + " error!");
                Console.WriteLine("  Using DEFAULT value: " + Max + " ");
                Console.WriteLine("  ");

                firstArg = Max;
            }
            Max = firstArg;

            while (i <= Max)
            {
                Console.WriteLine("Hello World ! " + i++ + " of " + Max);
            }
            Console.WriteLine(" ");

            return --i;
        }
    }
}
